### PR TITLE
`dbt run` works with Click

### LINKED ISSUE #5551: [CT-927] `dbt run` works in click

### PR DESCRIPTION
**Pull request by iknox-fa**
resolves #5551

### Description
`dbt run` works with Click.  Have fun folks!

Caveats:
* Integration tests are failing.  This is expected-- we'll get them sorted shortly in another PR.
* The "old" cli is officially broken by this PR.  Use the following command to run the new cli from the command line:
    `python3 -m dbt.cli.main run --project-dir ../path/to/project`
* If you wish to compare the old and new methods look through the commits-- `703b7e5` specifically contains both old and new code paths for the entire PR.
* This already has the changes in #6336 merged in.  If you wait until that's merged you'll have less to review here (if github updates the PR... I notice that doesn't always happen /shrug)

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: core/dbt/cli/params.py [hunk 1]
```diff
@@ -270,6 +262,15 @@
     default="default",
 )
 
+select = click.option(
```
Comment (ChenyuLInx) [inline core/dbt/cli/params.py:265]: like the new name

File: core/dbt/config/profile.py [hunk 2]
```diff
@@ -399,11 +399,13 @@ def from_raw_profiles(
         )
 
     @classmethod
-    def render_from_args(
+    def render(
```
Comment (ChenyuLInx) [inline core/dbt/config/profile.py:402]: I guess a lot of the diff will disapear once profile PR is in

File: core/dbt/task/debug.py [hunk 3]
```diff
@@ -71,6 +71,10 @@ def __init__(self, args, config):
             else:
                 self.project_dir = os.getcwd()
         self.project_path = os.path.join(self.project_dir, "dbt_project.yml")
+        # N.B. parse_cli_vars is embedded into the param when using click.
+        # replace this with:
```
Comment (ChenyuLInx) [inline core/dbt/task/debug.py:75]: might want to leave a note in the actual gh issue?

**Review by ChenyuLInx — APPROVED** [review]
Ops was intended to start review but ended up left a lot of comment, this looks good to me!

**Review by stu-k — APPROVED** [review]
Looks good! 👍

**Pull request merged**